### ISSUE TITLE
Enforce EOL Consistency Checks

### ISSUE THREAD

**Issue by tajmone** [editorconfig/editorconfig#431]
Although adding the `native` option for `end_of_line` has been already discussed and rejected (#226), it would be useful to add an option that checks that EOLs are consistent within the same file (regardless of whether it uses `CRLF`, `LF` or `CR`).

During the discussions in #226 (and elsewhere) it became quite clear that many users rely on EditorConfig for validating repositories' code consistency via Continuous Integration services like Travis, etc. 

Since Git defaults to treating `text` files EOLs as native (i.e. using `CRLF` on Windows and `LF` on Unix-like systems), the only viable solution to employ EditorConfig for code validation is to configure native sources to `end_of_line = unset` — which disable any type of EOL checks.

I think it would be useful if EditorConfig settings would allow to check for EOL consistency, so that even if `end_of_line` is `unset` code validators like [EClint] could at least check that the same EOL is being used in the file. Possible, something like:

    check_end_of_line_consistency = true

Mixed EOL types in a same file are a common inconsistency found in cross-platform collaborative projects, and can cause many problems. Since in Git project that contain sources set to `native` EditorConfig needs to use `end_of_line = unset` for those files, the result is that EditorConfig won't enforce EOL consistency on end users editors, leaving the door open for mixed EOLs introduction (i.e. the user's settings will prevail, adding whatever EOL type it considers appropriate).

It's true that Git can be forced to check for EOL consistency via some core settings, but a project can't safely rely on this and, besides, it would make sense to keep EditorConfig code validation as self-sufficient as possible.  

The new option proposed here would at least enforce consistent EOL usage. Possibly, this option will mostly affect code validating tools rather than editors — although some editors might enforce a similar option if a package is able to check/lint the source file for EOL incongruence.

> __NOTE__ — I'm strong supporter of adding the `end_of_line = native` option, but since this was already discussed and turned down I'm not bringing up the issue again; but I hope that this feature request could at least mitigate the current EditorConfig limitations for code validation by ensuring an EOL-agnostic consistency.

[EClint]: https://www.npmjs.com/package/eclint

**Comment by greut** [comment]
`end_of_line = consistent` sounds about right.

**Comment by tajmone** [comment]
Yes, it makes sense since it would be used as a fallback resort to `lf` or `crlf`. This way no new key needs to be created, just a new option/value. Well thought @greut!

**Comment by florianb** [comment]
How does EC know which line ending should be "enforced"?

**Comment by tajmone** [comment]
> How does EC know which line ending should be "enforced"?

First EOL sequence encountered should set the standard.

**Comment by greut** [comment]
@florianb what is hard is to enforce it "locally" when [Git permits you][autocrlf] to use different line endings locally and remotely. E.g. on Windows, I'd like to have some `crlf` to please my text editor, but once it's commited to Git, it's `lf` to please the rest of my team.

[autocrlf]: https://www.git-scm.com/book/en/v2/Customizing-Git-Git-Configuration#_core_autocrlf

**Comment by florianb** [comment]
Thanks @greut, what is the Git-config you're using? `autocrlf = true` with some kind of `text=auto`?
